04.ls/ls.rb

```diff
@@ -1,18 +1,42 @@
 #!/usr/bin/env ruby
 
-dir = Dir.glob '*'
-FILES = dir.count
-COLUMN = 3
-ROW = FILES.ceildiv(COLUMN)
-
-MAX_ROW = dir.max_by(&:length).length + 2
-
-index = 0
-1.upto(ROW) do |row| # 行ごとに繰り返し
-  COLUMN.times do # 列ごとに繰り返し
-    print dir[index].to_s.ljust(MAX_ROW, ' ') unless dir[index].nil?
-    index += ROW # 行数の倍数 == 配列から取得するindex変数の値
+require 'optparse'
+require 'etc'
+
+option = nil
+ARGV.options do |opt|
+  opt.on('-l') { option = '-l' }
+  opt.parse!
+end
+
+def l_option
+  Dir.glob('*').each do |name|
+    info = File.stat(name)
+    mode = info.mode.to_s(8)[-4..-1]
+    mode = mode.gsub('7', 'rwx').gsub('6', 'rw-').gsub('5', 'r-x').gsub('4', 'r--').gsub('3', '-wx').gsub('2', '-w-').gsub('1', '--x').tr('0', '-')
+    u_name = Etc.getpwuid(info.uid).name
+    group = Etc.getgrgid(info.gid).name
+    puts "#{mode}  #{info.nlink} #{u_name}  #{group}  #{info.size}  #{info.mtime.mon} #{info.mtime.day} #{info.mtime.hour}:#{info.mtime.min} #{name}"
+  end
+end
+
+option == '-l' ? l_option : dir = Dir.glob('*')
```

Comment:
`option` が `-l` かどうかを判定する処理が2箇所に分かれてしまっています。1箇所にまとめられないでしょうか？

同様に、 `Dir.glob('*')` も2箇所に分かれてしまっており、これだと `-a` オプションに対応しようとしたときに2箇所それぞれを修正することになります。1箇所にまとめていただきたいです 🙏

Reply:
下記のようにオプションの判定条件を一箇所に集約し、`dir = Dir.glob('*')`の部分も一箇所に集約しましたので、こちらで認識あっているかご確認願います。
```suggestion
dir = Dir.glob('*')
option == '-l' ? l_option(dir) : origin_ls_command(dir)
```